### PR TITLE
Mempool msg selection should respect block message limits

### PR DESCRIPTION
**Pull request by arajasek**

### CODE REVIEW

**Review by vyzo — COMMENTED** [review]
ugh, this looks ugly... not passing judgement yet though.

**Review by magik6k — APPROVED** [review]

File: chain/messagepool/selection.go [hunk 1]
```diff
@@ -46,24 +51,155 @@ func (mp *MessagePool) SelectMessages(ctx context.Context, ts *types.TipSet, tq
...
 
-	if len(msgs) > build.BlockMessageLimit {
-		msgs = msgs[:build.BlockMessageLimit]
+	if sm == nil {
+		return nil, nil
+	}
+
+	// one last sanity check
+	if len(sm.msgs) > build.BlockMessageLimit {
+		sm.msgs = sm.msgs[:build.BlockMessageLimit]
```
Comment (Kubuxu) [inline chain/messagepool/selection.go:72]: Log warning or error because this means that the code is wrong.

**Review by Kubuxu — APPROVED** [review]
SGTM, 1 nit

File: chain/messagepool/selection.go (hunk 1 above)
Comment (arajasek) [reply chain/messagepool/selection.go:72]: Yessir!

**Pull request merged**